Python/pythonrun.c

```diff
@@ -1485,6 +1485,70 @@ PyOS_CheckStack(void)
 
 #endif /* WIN32 && _MSC_VER */
 
+#if defined(__APPLE__)
+
+#define MAC_OS_10_9 13
+#define MAC_OS_10_10 14
+#define MAC_OS_10_11 15
+
+#include <sys/utsname.h>
+
+/*
+ * Return non-zero when we run out of memory on the stack; zero otherwise.
+ */
+
+__thread size_t last_remains = 0; /* Size of the last stack remain space */ 
+__thread size_t stack_space = 0; /* Size of a thread stack space */
```

Comment:
The CPython convention is to store thread-specific state in PyThreadState, please add those variables there, but see my comments on the use of these variables as well.